### PR TITLE
SSPROD-25956: TF support for Organizational Agentless scanning

### PR DESCRIPTION
**Pull request by ravinadhruve10**
Change summary:
----------------

1. For Agentless scanning, organizational installs will use CF to create stacksets and instances for :-

- IAM role and policy for agentless-scanning,
- KMS primary Key in each account, and each region,
- KMS primary Alias  in each account, and each region,

2. Minor updates to the KMS key module in main.tf
3. Updated the README and added appropriate comments in .tf files for agentless scanning.

Sample main.tf (Agentless scanning, Org Install) :-
```
provider "aws" {
  region = "us-east-1"
}

module "mgmt-account-role" {
  source           = "draios/secure-for-cloud/aws//modules/services/trust-relationship"
  role_name        = "sysdig-secure-2u6g"
  trusted_identity = "arn:aws:iam::0646xxxxxxxx:role/us-east-1-integration01-secure-assume-role"
  external_id      = "81145517xxxxxxxxxxxxxxxxxxxxxx"
}

module "organization-agentless-scanning" {
  source               = "draios/secure-for-cloud/aws//modules/services/agentless-scanning"
  trusted_identity     = "arn:aws:iam::0646xxxxxxxx:role/us-east-1-integration01-secure-assume-role"
  external_id          = "81145517xxxxxxxxxxxxxxxxxxxxxx"
  name                 = "sysdig-secure-scanning-i9yc"
  kms_key_alias        = "test"
  is_organizational    = true
  org_units            = ["r-op65"]
  instrumented_regions = ["us-east-1", "us-west-1", "us-west-2"]
}
```

### CODE REVIEW

**Comment by ravinadhruve10** [comment]
> Based on our discussion, these changes use `deploy_global_resources` in organizational install, to determine if the resource is to be created just once in primary region, or in additional/instrumented regions.
> 
> Running` terraform plan` generated following resources based on the sample main.tf below:- In primary region (us-east-1):
> 

As per the latest refactored changes, running `terraform plan` generated following resources, based on the sample main.tf below.

Sample main.tf :-
```
provider "aws" {
  region = "us-east-1"
}

module "organization-agentless-scanning-global" {
  source                  = "../agentless-scanning"
  trusted_identity        = "arn:aws:iam::0646xxxxxxxx:role/us-east-1-integration01-secure-assume-role"
  external_id             = "81145517xxxxxxxxxxxxxxxxxxxxxx"
  name                    = "sysdig-secure-scanning-i9yc"
  kms_key_alias           = "test"
  deploy_global_resources = true
  is_organizational       = true
  org_units               = ["r-op65"]
}

module "organization-agentless-scanning-non-global" {
  source               = "../agentless-scanning"
  trusted_identity     = "arn:aws:iam::0646xxxxxxxx:role/us-east-1-integration01-secure-assume-role"
  external_id          = "81145517xxxxxxxxxxxxxxxxxxxxxx"
  name                 = "sysdig-secure-scanning-i9yc"
  kms_key_alias        = "test"
  is_organizational    = true
  org_units            = ["r-op65"]
  instrumented_regions = ["us-west-2", "us-west-1"]
}
```

Single globally created resources :-

- IAM Policy
- IAM Role
- Role Attachment
- IAM Role + Policy Stackset for agentless scanning
- IAM Role + Policy Stackset instance (one per account in the org)

In management account, global resources in primary region (us-east-1):

- (KMS Primary key + Alias) Stackset
- (KMS Primary key + Alias) Stackset instance (one in management account)

In management account, non-global resources per instrumented region (us-west-2, us-west-1):

- (KMS Replica key + Alias) Stackset
- (KMS Replica key + Alias) Stackset instance (one per instrumented region in management account)


In each child account, global resources in primary region (us-east-1):

- (KMS Primary key + Alias) Stackset
- (KMS Primary key + Alias) Stackset instance (one per account in the org)

In each child account, non-global resources per instrumented region (us-west-2, us-west-1):

- (KMS Replica key + Alias) Stackset
- (KMS Replica key + Alias) Stackset instance (one per account in the org, per each instrumented region)

**Comment by ravinadhruve10** [comment]
Latest status on this PR is described in the JIRA ticket here - https://sysdig.atlassian.net/browse/SSPROD-25956

**Comment by ravinadhruve10** [comment]
> Latest status on this PR is described in the JIRA ticket here - https://sysdig.atlassian.net/browse/SSPROD-25956

Updated Sample main.tf for reference, as per latest discussions :-

```
provider "aws" {
  region = "us-east-1"
}

module "organization-agentless-scanning" {
  source               = "../agentless-scanning"
  trusted_identity     = "arn:aws:iam::0646xxxxxxxx:role/us-east-1-integration01-secure-assume-role"
  external_id          = "81145517xxxxxxxxxxxxxxxxxxxxxx"
  name                 = "sysdig-secure-scanning-i9yc"
  kms_key_alias        = "test"
  is_organizational    = true
  org_units            = ["r-op65"]
  instrumented_regions = ["us-east-1", "us-west-1", "us-west-2"]
}
```

**Review by cgeers — DISMISSED** [review]
great job!

**Comment by cgeers** [comment]
@ravinadhruve10 I've pushed a small commit to that removes the use of replica keys, and updates documentation. Please give this commit a look and merge at your discretion.

File: modules/services/agentless-scanning/README.md [hunk 1]
```diff
@@ -29,12 +29,19 @@ No modules.
 
 | Name | Type |
 |------|------|
+| [aws_cloudformation_stack_set.mgmt_acc_resources_stackset](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/cloudformation_stack_set) | resource |
+| [aws_cloudformation_stack_set.ou_resources_stackset](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/cloudformation_stack_set) | resource |
+| [aws_cloudformation_stack_set.scanning_role_stackset](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/cloudformation_stack_set) | resource |
+| [aws_cloudformation_stack_set_instance.mgmt_acc_global_stackset_instance](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/cloudformation_stack_set_instance) | resource |
```
Comment (ravinadhruve10) [inline modules/services/agentless-scanning/README.md:35]: These 4 stackset_instances (`mgmt_acc_global_stackset_instance`, `mgmt_acc_nonglobal_stackset_instance`, `ou_global_stackset_instance`, `ou_nonglobal_stackset_instance`) should be consolidated into 2 actually - `mgmt_acc_stackset_instance`, `ou_global_stackset_instance`

**Comment by ravinadhruve10** [comment]
> great job!

Thank you! :)

I have a minor change in this PR, will push that along with the README.md fix.

File: modules/services/agentless-scanning/README.md (hunk 1 above)
Comment (ravinadhruve10) [reply modules/services/agentless-scanning/README.md]: Fixed.

**Review by cgeers — DISMISSED** [review]
nice catch!

**Review by cgeers — APPROVED** [review]

**Pull request merged**